src/Umbraco.Core/Services/MediaEditingService.cs

```
@@ -51,42 +48,59 @@ internal sealed class MediaEditingService
     }
 
-    public async Task<Attempt<IMedia?, ContentEditingOperationStatus>> MoveToRecycleBinAsync(Guid id, Guid userKey)
-    {
-        var currentUserId = await GetUserIdAsync(userKey);
-        return await HandleDeletionAsync(id, media => ContentService.MoveToRecycleBin(media, currentUserId).Result, false);
-    }
+    public async Task<Attempt<IMedia?, ContentEditingOperationStatus>> MoveToRecycleBinAsync(Guid key, Guid userKey)
+        => await HandleMoveToRecycleBinAsync(key, userKey);
 
-    public async Task<Attempt<IMedia?, ContentEditingOperationStatus>> DeleteAsync(Guid id, Guid userKey)
-    {
-        var currentUserId = await GetUserIdAsync(userKey);
-        return await HandleDeletionAsync(id, media => ContentService.Delete(media, currentUserId).Result, true);
-    }
+    public async Task<Attempt<IMedia?, ContentEditingOperationStatus>> DeleteAsync(Guid key, Guid userKey)
+        => await HandleDeleteAsync(key, userKey);
+
+    public async Task<Attempt<IMedia?, ContentEditingOperationStatus>> MoveAsync(Guid key, Guid? parentKey, Guid userKey)
+        => await HandleMoveAsync(key, parentKey, userKey);
 
-    public async Task<Attempt<IMedia?, ContentEditingOperationStatus>> MoveAsync(Guid id, Guid? parentId, Guid userKey)
+    public async Task<ContentEditingOperationStatus> SortAsync(Guid? parentKey, IEnumerable<SortingModel> sortingModels, Guid userKey)
+        => await HandleSortAsync(parentKey, sortingModels, userKey);
+
+    protected override IMedia New(string? name, int parentId, IMediaType mediaType)
+        => new Models.Media(name, parentId, mediaType);
+
+    protected override OperationResult? Move(IMedia media, int newParentId, int userId)
+        => ContentService.Move(media, newParentId, userId).Result;
+
+    protected override IMedia? Copy(IMedia media, int newParentId, bool relateToOriginal, bool includeDescendants, int userId)
+        => null; // copy is not supported for media
```

``` suggestion
        => throw new NotImplementedException("Copy is not supported for media");
```
I like just failing fast here 🚀

---

I'm all for it 💥

---

I would prefer NotSupportedException. 

NotImplementedException is typically highlighted like TODOs in IDEs

---

I can't amend it at this point, I'm sat with a trillion changes in another branch. @Zeegaan can you update the code before merging?

---

I guess github can help here :), I'll take a look

---

Oh, need to update OpenAPI.json anyway